### PR TITLE
Arrays and linked list solutions rich

### PR DESCRIPTION
**Pull request by rashworth7**
My solutions for arrays and linked lists

### CODE REVIEW

File: exercises/k9-data-structures/lists/linked-list-vs-array/linked-list/problem.js [hunk 1]
```diff
@@ -20,9 +20,83 @@ export class LinkedList {
...
+
+        return pointer.value
+    }
+    insertAt(index, value) {
+
+        if (index > this.size()) {
+            throw new Error ('index out of bounds');
+        }
+
+        if (!this.head) {
```
Comment (jonhawkinsctm) [inline exercises/k9-data-structures/lists/linked-list-vs-array/linked-list/problem.js]: This stops appending to an empty list... admittedly there's not a test for that tho!

File: exercises/k9-data-structures/lists/linked-list-vs-array/shared/test-suite.js [hunk 2]
```diff
@@ -7,7 +7,8 @@ export function ListTestSuite(factory) {
             expect(list.get(0)).toBe(0);
             expect(list.get(1)).toBe(1);
             expect(list.get(2)).toBe(2);
-            expect(list.get(3)).toBe(null);
+            expect(() => { list.get(3) }).toThrow();
```
Comment (jonhawkinsctm) [inline exercises/k9-data-structures/lists/linked-list-vs-array/shared/test-suite.js]: You need to run the array tests again, I'm not sure they throw 😉 

If we're changing the contract to throw, we should make assertions about what that error is. Maybe we should have throw a specific error when accessing an undefined index: `expect(() => { list.get(3) }).toThrow("index out of bounds")`. That said throwing on accessing a value can be dangerous if there is no way to check it exists–if you wanted to take this approach further try implementing a `indexExists` function allow consumers to check before accessing.

File: exercises/k9-data-structures/lists/linked-list-vs-array/linked-list/problem.js [hunk 3]
```diff
@@ -20,9 +20,83 @@ export class LinkedList {
...
+            this.head = new Node(value);
+        }
+
+        let pointer = this.head;
+        while (pointer.next) {
+            pointer = pointer.next
+        }
+
+        pointer.next = new Node (value);
```
Comment (jonhawkinsctm) [inline exercises/k9-data-structures/lists/linked-list-vs-array/linked-list/problem.js]: Nitpicking, but there shouldn't be white space between the name and the parentheses when calling a function/constructor.

File: exercises/k9-data-structures/lists/linked-list-vs-array/array/problem.js [hunk 4]
```diff
@@ -9,9 +9,37 @@ export class MyArray {
...
+    get(index) {
+        if (index < this.length) {
+            return this.nodes[index];
+        }
+        return null;
+    }
+    insertAt(index, value) {
+        const temp = { ...this.nodes }
+        this.nodes[index] = value
+        for (let i = index + 1; i < this.length + 1; i++) {
```
Comment (jonhawkinsctm) [inline exercises/k9-data-structures/lists/linked-list-vs-array/array/problem.js:36]: For extra points, there is a way to make this loop shorter if the index is in the middle of the array 😉 Consider inserting `"A"` into `[1,2,3,4,5]` to get `[1,2,3,"A",4,5]`... which indexes do we _need_ to update?

**Review by jonhawkinsctm — COMMENTED** [review]
Got a few comments for consideration/extra thinking, but on the whole really good!

File: exercises/k9-data-structures/lists/linked-list-vs-array/array/problem.js (hunk 4 above)
Comment (rashworth7) [reply exercises/k9-data-structures/lists/linked-list-vs-array/array/problem.js:36]: Just the final 2?

File: exercises/k9-data-structures/lists/linked-list-vs-array/shared/test-suite.js (hunk 2 above)
Comment (rashworth7) [reply exercises/k9-data-structures/lists/linked-list-vs-array/shared/test-suite.js]: Can you run through this with me quickly at some point? I thought I had checked the test, clearly not 🙈

File: exercises/k9-data-structures/lists/linked-list-vs-array/array/problem.js (hunk 4 above)
Comment (jonhawkinsctm) [reply exercises/k9-data-structures/lists/linked-list-vs-array/array/problem.js:36]: my bad

**Pull request merged**